src/Microsoft.TemplateEngine.Edge/BuiltInManagedProvider/GlobalSettings.cs

```diff
@@ -135,10 +132,10 @@ public async Task SetInstalledTemplatePackagesAsync(IReadOnlyList<TemplatePackag
                 try
                 {
                     // Ignore FSW notifications received during writing changes (we'll notify synchronously)
-                    Interlocked.Increment(ref _waitingInstances);
+                    _watcher?.Dispose();
```

Comment:
https://learn.microsoft.com/en-us/dotnet/api/system.io.filesystemwatcher.enableraisingevents?view=net-7.0 

won't be easier to disable events?